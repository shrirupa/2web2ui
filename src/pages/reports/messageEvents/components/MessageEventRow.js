/* eslint-disable max-lines */
import React, { Component } from 'react';
import { snakeToFriendly } from 'src/helpers/string';
import DisplayDate from 'src/components/displayDate/DisplayDate';
import { getDetailsPath } from 'src/helpers/messageEvents';
import _ from 'lodash';
import styles from './MessageEventRow.module.scss';
import { Table } from '@sparkpost/matchbox';
import { withRouter } from 'react-router-dom';

export class MessageEventRow extends Component {
  onClick = () => {
    const url = getDetailsPath(this.props.event.message_id, this.props.event.event_id);
    this.props.history.push(url);
  };

  render() {
    return (
      <Table.Row className={styles.Row} onClick={this.onClick}>
        <Table.Cell className={styles.Cell}>{snakeToFriendly(this.props.event.type)}</Table.Cell>
        <Table.Cell className={styles.Cell}>{this.props.event.subject}</Table.Cell>
        <Table.Cell className={styles.Cell}>{this.props.event.rcpt_to}</Table.Cell>
        <Table.Cell className={styles.Cell}>{this.props.event.friendly_from}</Table.Cell>
        <Table.Cell className={styles.Cell}><DisplayDate timestamp={this.props.event.timestamp} formattedDate={this.props.event.formattedDate} /></Table.Cell>
      </Table.Row>
    );
  }
}

export default withRouter(MessageEventRow);
