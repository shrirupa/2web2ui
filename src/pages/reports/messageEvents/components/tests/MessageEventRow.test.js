import React from 'react';
import { shallow } from 'enzyme';
import { MessageEventRow } from '../MessageEventRow';

describe('Component: MessageEventRow', () => {
  const props = {
    event: {
      formattedDate: 'formatted',
      type: 'injection',
      friendly_from: 'mean@friendly',
      rcpt_to: 'tom.haverford@pawnee.state.in.us',
      message_id: '123abc',
      event_id: '456xyz',
      history: {
        push: jest.fn()
      }
    }
  };

  it('renders correctly', () => {
    expect(shallow(<MessageEventRow {...props} />)).toMatchSnapshot();
  });
});
