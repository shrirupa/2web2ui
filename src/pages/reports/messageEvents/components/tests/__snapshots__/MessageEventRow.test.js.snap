// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MessageEventRow renders correctly 1`] = `
<Table.Row
  className="Row"
  onClick={[Function]}
>
  <Table.Cell
    className="Cell"
  >
    Injection
  </Table.Cell>
  <Table.Cell
    className="Cell"
  />
  <Table.Cell
    className="Cell"
  >
    tom.haverford@pawnee.state.in.us
  </Table.Cell>
  <Table.Cell
    className="Cell"
  >
    mean@friendly
  </Table.Cell>
  <Table.Cell
    className="Cell"
  >
    <DisplayDate
      formattedDate="formatted"
    />
  </Table.Cell>
</Table.Row>
`;
